File: LeetCode/p03_longest_substring_without_repeating_characters/program.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        result = 0
        hash_map = {}

        i = 0    # left pointer
        j = 0    # right pointer

        while j < len(s):
            ch = s[j]

            if ch in hash_map:
                i = max(hash_map[ch], i)

            result = max(result, j - i + 1)
            hash_map[ch] = j + 1

            j += 1
        return result

File: LeetCode/p03_longest_substring_without_repeating_characters/program.kt
class Solution {
    fun lengthofSubString(str: String): Int{
        var result = 0
        var i = 0
        var j = 0

        var hashMap = HashMap<Char,Int>()

        while(j < str.length){
            val ch = str[j]

            if(hashMap.containsKey(ch)){
                i = Math.max(hashMap[ch]!!, i)
            }

            result = Math.max(result, j - i + 1)
            hashMap[ch] = j + 1

            j++;
        }

        return result
    }
}